### ISSUE TITLE
Wrong logic is used in case of events

### ISSUE THREAD

**Issue by kkamkou** [FutoRicky/ember-cli-dropzonejs#21]
Events are used to `notify` other listeners about an event. What you did, is just a wrapper for a new object.

What was expected:
- drop-zone emits an `event`, I can `subscribe` and react

What we have:
- `{{drop-zone url=imageUploadUrl addRemoveLinks=true complete=(action "onUploadComplete")}}`
- The result is: I've created a new object with MY OWN `complete` event, replacing the default one.

How to fix:
- In case of events, you should register a callback using the `emit` function. Preserving the native event. Example: `instance.on('complete', this.complete)`

Workaround:
- Use only `noop` events, like: `queuecomplete`, `addedfiles`, etc. [see the source code](https://github.com/enyo/dropzone/blob/master/src/dropzone.coffee).

**Comment by kmiyashiro** [comment]
Yes, it seems that you are actually overriding the hooks instead of adding event listeners. Maybe adding properties like `onaddedfile` etc for adding event listeners to dropzone events would work. I think this problem is compounded by the fact that access to the Dropzone instance is impossible by actions unless you extend the component.

**Comment by FutoRicky** [comment]
I am having trouble understanding the issue. How would I go to recreate this problem?

Also, @kkamkou try assigning a function property instead of an action to the `complete attribute`.

**Comment by kkamkou** [comment]
> try assigning a function property instead of an action

It doesn't matter. In the variant you propose me, I lose the context of the callback. And it won't fix the problem.

> Yes, it seems that you are actually overriding the hooks instead of adding event listeners.

true

> I am having trouble understanding the issue.

I see it from the code. All you need to do (in case you want to fix it) is to register events like events and not like an attribute. `forEach() if (this.complete is function) {dropzone.on('complete', this.complete)}`

**Comment by FutoRicky** [comment]
@kkamkou oohh yes I see. Thanks for pointing this out. Will definitely fix this.

**Comment by FutoRicky** [comment]
Fixed the event handling logic 7eb7b33a66f672170345dab849d18e2b606929cb. 

Declare the event function in controller for example:

```
addedFileEvent: Ember.computed(function() {
  return function() {
    // do something...
  };
}),
```

Declare component

`{{drop-zone url=http://example.com addfile=addedFileEvent}}`

You can update to version `0.7.0` to have this fix

**Comment by kkamkou** [comment]
It works, thank you.
